### PR TITLE
[ALLUXIO-3243] Remove sync from acquireUfsResource

### PR DESCRIPTION
**Pull request by gpang**
https://alluxio.atlassian.net/browse/ALLUXIO-3243

### CODE REVIEW

**Comment by gpang** [comment]
This change reduces some of the contention on the `synchronized` method of getting a ufs resource.

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

File: core/server/common/src/main/java/alluxio/underfs/UfsManager.java [hunk 1]
```diff
@@ -43,17 +44,18 @@ public UfsClient(Supplier<UnderFileSystem> ufsSupplier, AlluxioURI ufsMountPoint
...
 
     /**
      * @return the UFS instance
      */
-    public synchronized CloseableResource<UnderFileSystem> acquireUfsResource() {
-      if (mUfs == null) {
-        mUfs = mUfsSupplier.get();
+    public CloseableResource<UnderFileSystem> acquireUfsResource() {
+      if (mUfs.get() == null) {
+        mUfs.compareAndSet(null, mUfsSupplier.get());
```
Comment (apc999) [inline core/server/common/src/main/java/alluxio/underfs/UfsManager.java]: unsafe here, better to check the result of cas operation and close resource accordingly.
```
UnderFileSystem ufs = mUfsSupplier.get();
if (!mUfs.compareAndSet(null, ufs)) ufs.close();
```

Comment (gpang) [reply core/server/common/src/main/java/alluxio/underfs/UfsManager.java]: Added close. Thanks!

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
@madanadit

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Pull request merged**